crates/dojo-world/src/migration/strategy.rs

```diff
@@ -105,12 +108,32 @@ where
 
     // If the world contract needs to be migrated, then all contracts need to be migrated
     // else we need to evaluate which contracts need to be migrated.
-    let world = evaluate_contract_to_migrate(&diff.world, &artifact_paths, false)?;
-    let executor = evaluate_contract_to_migrate(&diff.executor, &artifact_paths, world.is_some())?;
+    let mut world = evaluate_contract_to_migrate(&diff.world, &artifact_paths, false)?;
+    let mut executor =
+        evaluate_contract_to_migrate(&diff.executor, &artifact_paths, world.is_some())?;
     let components =
         evaluate_components_to_migrate(&diff.components, &artifact_paths, world.is_some())?;
     let systems = evaluate_systems_to_migrate(&diff.systems, &artifact_paths, world.is_some())?;
 
+    if let Some(executor) = &mut executor {
+        executor.contract_address =
+            get_contract_address(FieldElement::ZERO, diff.executor.local, &[], FieldElement::ZERO);
+    }
+
+    // If world needs to be migrated, then we expect the `seed` to be provided.
+    if let Some(world) = &mut world {
+        let salt =
+            seed.map(poseidon_hash_single).ok_or(anyhow!("Missing seed for World deployment."))?;
```

Comment:
```suggestion
            seed.map(poseidon_hash_single).ok_or(anyhow!("Missing seed for World deployment."))?;
```

This error message could be confusing to users, since we refer to it as `name` in the cli. perhaps we should use `name` here too for consistency?

Reply:
i tried to make the error message based on the context of the function.